base_classes/NXdetector.nxdl.xml

```diff
@@ -424,6 +424,19 @@
         <dim index="3" value="k" />
       </dimensions>
     </field>
+  </group>
+
+  <field name="real_time" type="NX_NUMBER" units="NX_TIME">
+    <doc>
+      real-time of the exposure (use this if exposure time varies for
+      each array element, otherwise use ``count_time`` field)
+    </doc>
+    <dimensions rank="3">
```

Comment:
Should this really be 3D?

Reply:
Most likely not and 1D is enough. After all we just want a number per frame.

Reply:
The question is: _Is there such a detector that can vary the time across the frame?_  I believe the answer is _yes_ when looking at fast event-counting (so called fast pixel) detectors which a frame represents a number of disparate counting events.  It is not realistic to gather the many individual events from such a detector as raw data.  Rather, the composite frame is accumulated from the event stream.

Reply:
@ronpandolfi: Is the fast pixel detector at ALS such an example?

Reply:
@prjemian Let's put it like this. Can one use a 1D dataset when rank is set to 3 in the definition?

Reply:
I agree that a 1-D should be allowed.  That's the most typical case.  How do we describe this in NXDL?  How is it described in other base classes when a variable rank array is allowed?

Reply:
I guess it will be the same as for the polar angle:

``
This is the polar angle of the detector towards the previous component in the instrument; most often the sample. The usage depends on the nature of the detector. Most often it is the polar_angle of the detector assembly. But there are irregular detectors. In this case, the polar_angle must be specified for each detector pixel.
``

We could state a similar syntax and sentence here:

``
Most often there is a real time for the detector signal but there are detectors in which the real time changes per pixel and in that case more than one dimension is needed.
``

Reply:
So should be <= detector rank? (Still not 3D.)